From 22d31cdc21523aa790c07d84ee1b89dcb53107af Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Timo=20Ter=C3=A4s?= <timo.teras@iki.fi>
Date: Wed, 30 Mar 2016 13:46:57 +0300
Subject: [PATCH 1/5] zebra: use FIB state for nexthop tracking

The FIB override routes can override ZEBRA_FLAG_SELECTED routes
in FIB. Use the FIB state instead to report correct nexthop when
FIB override routes are present.
---
 zebra/zebra_rnh.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/zebra/zebra_rnh.c b/zebra/zebra_rnh.c
index 7b6d0f8..97d3597 100644
--- a/zebra/zebra_rnh.c
+++ b/zebra/zebra_rnh.c
@@ -223,7 +223,7 @@ zebra_evaluate_rnh_table (vrf_id_t vrfid, int family)
 	    {
 	      if (CHECK_FLAG (rib->status, RIB_ENTRY_REMOVED))
 		continue;
-	      if (CHECK_FLAG (rib->flags, ZEBRA_FLAG_SELECTED))
+	      if (CHECK_FLAG (rib->status, RIB_ENTRY_SELECTED_FIB))
 		{
 		  if (CHECK_FLAG(rnh->flags, ZEBRA_NHT_CONNECTED))
 		    {
-- 
2.10.1

